validator/src/admin_rpc_service.rs

```diff
@@ -1012,4 +1071,183 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_get_largest_index_keys() {
+        // Constants
+        const NUM_DUMMY_ACCOUNTS: usize = 50;
+        const MAX_CHILD_ACCOUNTS: usize = 5; // Set low because it induces lots of same key size entries in the ProgramID list
+        const MAX_MINT_ACCOUNTS: usize = 50;
+        const MAX_TOKEN_ACCOUNTS: usize = 100;
+
+        // Set secondary indexes
+        let account_indexes = AccountSecondaryIndexes {
+            keys: None,
+            indexes: HashSet::from([
+                AccountIndex::ProgramId,
+                AccountIndex::SplTokenMint,
+                AccountIndex::SplTokenOwner,
+            ]),
+        };
+
+        // RPC & Bank Setup
+        let rpc = RpcHandler::start_with_config(TestConfig { account_indexes });
+
+        let bank = rpc.root_bank();
+        let RpcHandler { io, meta, .. } = rpc;
+
+        // Add some basic system owned account
+        let mut dummy_account_pubkeys = Vec::with_capacity(NUM_DUMMY_ACCOUNTS);
+        let mut num_generator = thread_rng();
+        let key_size_range = Uniform::new_inclusive(0, MAX_CHILD_ACCOUNTS);
+        for _i in 1..=NUM_DUMMY_ACCOUNTS {
+            let pubkey = Pubkey::new_unique();
+            dummy_account_pubkeys.push(pubkey);
+            let account = AccountSharedData::from(Account {
+                lamports: 11111111,
+                owner: system_program::id(),
+                ..Account::default()
+            });
+            bank.store_account(&pubkey, &account);
+        }
+
+        // Now add a random number of accounts each owned by one of the newely
+        // created dummy accounts
+        for dummy_account in &dummy_account_pubkeys {
+            // Add child accounts to each dummy account
+            let num_children = (&mut num_generator).sample_iter(key_size_range).next();
+            for j in 0..num_children.unwrap_or(0) {
+                let child_pubkey = Pubkey::new_unique();
+                let child_account = AccountSharedData::from(Account {
+                    lamports: ((j as u64) + 1) * 1000,
```

Comment:
I see that I missed this in your other PR, but these won't necessarily all be rent-exempt. I don't see that the lamports needs to vary for any particular reason, so I'd recommend just adding this somewhere above to use here and on L1107:
```
let lamports = bank.get_minimum_balance_for_rent_exemption(0);
```

Reply:
It was for my own debugging initially. Gave me a differentiator on child accounts to verify the logic. Can probably remove now.